src/v/transform/logging/rpc_client.cc

```diff
@@ -0,0 +1,111 @@
+
+#include "cluster/topic_table.h"
+#include "model/namespace.h"
+#include "transform/logging/logger.h"
+#include "transform/logging/record_batcher.h"
+
+#include <seastar/core/future.hh>
+
+namespace transform::logging {
+
+rpc_client::rpc_client(
+  rpc::client* rpc_client, cluster::topic_table* topic_table)
+  : _rpc_client(rpc_client)
+  , _topic_table(topic_table) {}
+
+ss::future<errc>
+rpc_client::write(model::partition_id pid, io::json_batches batches) {
+    model::topic_namespace_view ns_tp{model::transform_log_internal_nt};
+    // TODO(oren): is topic metadata cache OR topic table more preferable here?
+    auto cfg = _rpc_client->find_topic_cfg(ns_tp);
+    if (!cfg.has_value()) {
+        co_return errc::topic_not_found;
+    }
+
+    // topic should reflect cluster config at time of creation, so we can assume
+    // batch_max_bytes it's set
+    auto max_batch_size = cfg->properties.batch_max_bytes;
+    vassert(
+      max_batch_size.has_value(),
+      "Missing max batch size property on {}",
+      ns_tp);
+
+    record_batcher batcher{*max_batch_size};
+
+    while (!batches.empty()) {
+        auto json_batch = std::move(batches.front());
+        batches.pop_front();
+        while (!json_batch.events.empty()) {
+            auto ev = std::move(json_batch.events.front());
+            json_batch.events.pop_front();
+            batcher.append(json_batch.name.copy(), std::move(ev));
+        }
+    }
+
+    auto total_request_size = batcher.total_size_bytes();
+    auto record_batches = batcher.finish();
+
+    model::topic_partition tp{model::transform_log_internal_topic, pid};
+    vlog(
+      tlg_log.debug,
+      "Producing logs for {{ {} }} total size: {}B",
+      tp,
+      total_request_size);
+
+    auto res = co_await _rpc_client->produce(
+      std::move(tp), std::move(record_batches));
+    if (res != cluster::errc::success) {
+        co_return errc::write_failure;
+    }
+    co_return errc::success;
+}
+
+result<model::partition_id, errc>
+rpc_client::compute_output_partition(model::transform_name_view name) {
+    model::topic_namespace_view ns_tp{model::transform_log_internal_nt};
+    const auto& config = _topic_table->get_topic_cfg(ns_tp);
+    if (!config) {
+        return errc::topic_not_found;
+    }
+
+    // TODO(oren): The disabled set dance might be unnecessary since the logs
+    // topic is not a "user_topic". I believe that excludes it from manual
+    // partition enable/disable. That could break our dependency on the
+    // topic table here (can use rpc::client metadata cache for lookup
+    // instead?), but perhaps it's a better idea to leave this in place?
+    const auto* disabled_set = _topic_table->get_topic_disabled_set(ns_tp);
+    if (!(disabled_set && disabled_set->is_fully_disabled())) {
+        for (size_t i = 0; i < static_cast<size_t>(config->partition_count);
+             ++i) {
+            model::partition_id candidate(
+              (sstring_hash{}(name()) + i)
```

Comment:
Oh something I forgot to catch here. Absl hashing is not a persistent hash. Another way to say that is that it is seeded differently for every process. Let's use `mumur2` or something.

Consider: https://github.com/redpanda-data/redpanda/blob/510f4593ac4354f54d5bb9e47f100feb8b474830/src/v/cluster/distributed_kv_stm.h#L204-L205

Reply:
nice one, thanks

Reply:
> Oh something I forgot to catch here. Absl hashing is not a persistent hash. Another way to say that is that it is seeded differently for every process. Let's use mumur2 or something.

yikes. that's an easy and devastating mistake to make